# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546406014 28800
#      Tue Jan 01 21:13:34 2019 -0800
# Branch stable
# Node ID 768945ca2ff26f8e2d8e286a4ca00ad1d58efa29
# Parent  bb3ebaa9460e4abfac4ed5e8a12290500232abd7
quad.cc: Fix static analyzer detected issues (bug #55347).

* quad.cc (Fquad): Call args(4).xvector_value()
rather than incorrect version without prepended 'x'.

diff --git a/libinterp/corefcn/quad.cc b/libinterp/corefcn/quad.cc
--- a/libinterp/corefcn/quad.cc
+++ b/libinterp/corefcn/quad.cc
@@ -339,17 +339,17 @@ variable by routines @code{dblquad} and 
       switch (nargin)
         {
         case 5:
           if (indefinite)
             error ("quad: singularities not allowed on infinite intervals");
 
           have_sing = true;
 
-          sing = args(4).vector_value ("quad: fifth argument SING must be a vector of singularities");
+          sing = args(4).xvector_value ("quad: fifth argument SING must be a vector of singularities");
           OCTAVE_FALLTHROUGH;
 
         case 4:
           tol = args(3).xvector_value ("quad: TOL must be a 1 or 2-element vector");
 
           switch (tol.numel ())
             {
             case 2:
